build_tools/cmake/iree_llvm.cmake

```diff
@@ -96,11 +104,13 @@ macro(iree_llvm_configure_installed)
     set(LLVM_EXTERNAL_LIT "${IREE_SOURCE_DIR}/third_party/llvm-project/llvm/utils/lit/lit.py")
   endif()
 
```

Comment:
Should be able to add

```
set(IREE_LLVM_LINK_TARGET llvm-link)
set(IREE_LLD_TARGET lld)
set(IREE_CLANG_TARGET clang)
```

If flexibility is necessary w/ different variants of byo_llvm, then gate these like 

```
set(IREE_CLANG_TARGET)
if(TARGET clang)
    set(IREE_CLANG_TARGET clang)
endif()
```

Reply:
Added.